common/namespace/cache.go

```diff
@@ -385,15 +373,7 @@ func (c *namespaceCache) refreshLoop() {
 	}
 }
 
-func (c *namespaceCache) refreshNamespaces() error {
-	c.refreshLock.Lock()
-	defer c.refreshLock.Unlock()
-	return c.refreshNamespacesLocked()
-}
-
-// this function only refresh the namespaces in the v2 table
-// the namespaces in the v1 table will be refreshed if cache is stale
-func (c *namespaceCache) refreshNamespacesLocked() error {
+func (c *namespaceCache) refreshNamespaces(ctx context.Context) error {
```

Comment:
why is lock no longer needed?

Reply:
Only ever called from 2 places: (1) in Start() where it completes before the refreshLoop is started, and (2) in the refreshLoop which is a single goroutine. There's no path to concurrent execution.